capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -90,22 +102,86 @@
     }
   }
 
+  public HashMap<String, Object> runViewStudySetSqlQuery(
+    DataSource pool,
+    String studySetID
+  )
+    throws SQLException {
+    HashMap<String, Object> studySetDetails = new HashMap<>();
+    ArrayList<Card> studySetCards = new ArrayList<>();
+
+    try (Connection conn = pool.getConnection()) {
+      try (
+        PreparedStatement query_statement = conn.prepareStatement(
+          VIEW_STUDY_SET_QUERY
+        )
+      ) {
+        studySetID = studySetID.substring(1);
+        query_statement.setInt(1, Integer.parseInt(studySetID));
+        ResultSet result = query_statement.executeQuery();
+        while (result.next()) {
+          if (studySetDetails.containsKey("title") == false) {
+            String title = result.getString("title");
+            studySetDetails.put("title", title);
+
+            String description = result.getString("description");
+            studySetDetails.put("description", description);
+
+            String subject = result.getString("subject");
+            studySetDetails.put("subject", subject);
+
+            String user_author = result.getString("user_name");
+            studySetDetails.put("user_author", user_author);
+
+            String university = result.getString("name");
+            studySetDetails.put("university", university);
+          }
+          String cardFront = result.getString("front");
+          String cardBack = result.getString("back");
+          Card card = new Card(cardFront, cardBack);
+          studySetCards.add(card);
+        }
+        studySetDetails.put("cards", studySetCards);
+        return studySetDetails;
+      }
+    }
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
     ServletContext servletContext = getServletContext();
-    String search_word  = request.getParameter("stringToSearchBy");
     DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
+    if (request.getPathInfo() == null) {
+      String search_word = request.getParameter("stringToSearchBy");
```

Comment:
This feels like a good place to extract a helper function, one function to get a String representing the search results and another function for the single result. This should also help remove a bit of duplication, e.g. only calling `response.getWriter().println` and `response.setContentType("application/json;")` in one place.

Reply:
> `response.getWriter().println` and `response.setContentType("application/json;")`

@ljfranklin  I made a helper function and uncluttered the doGet